### PR TITLE
Feature/city input

### PR DESCRIPTION
**Pull request by pasternakewa**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,63 @@
...
+import "./App.css";
+import React, { useState, useEffect } from "react";
+import Icon from "./components/Icon";
+
+const getQueryUrl = (city) =>
+  `${process.env.REACT_APP_API_URL}?q=${city}&units=metric&appid=${process.env.REACT_APP_API_KEY}`;
 
 function App() {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState("");
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa initialize as undefined

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,63 @@
...
+  const [weatherData, setWeatherData] = useState();
+  const [cityInput, setCityInput] = useState("Warszawa");
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa Let's use current location to initial city

File: src/App.js [hunk 3]
```diff
@@ -1,23 +1,63 @@
...
+  const [city, setCity] = useState(cityInput);
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa WOT? why You need this state?

File: src/App.js [hunk 4]
```diff
@@ -1,23 +1,63 @@
...
+
+  useEffect(() => {
+    setLoading(true);
+    fetch(getQueryUrl(city))
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa Please move this to separate function

File: src/App.js [hunk 5]
```diff
@@ -1,23 +1,63 @@
...
+        setError("fetch failed");
+      });
+  }, [city]);
+
+  if (loading) {
+    return <p>loading..</p>;
+  }
+
+  if (error) {
+    return <p>ERROR: {error}</p>;
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa create more explicit error here

File: src/App.js [hunk 6]
```diff
@@ -1,23 +1,63 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <input
+        value={cityInput}
+        onChange={(e) => setCityInput(e.target.value)}
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa move to separate function

File: src/App.js [hunk 7]
```diff
@@ -1,23 +1,63 @@
...
+        onKeyPress={(e) => handleKeypress(e)}
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa move to separate function

File: src/App.js [hunk 8]
```diff
@@ -1,23 +1,63 @@
...
+      />
+      <button onClick={() => setCity(cityInput)}>Submit</button>
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa move to separate function

File: src/App.js [hunk 9]
```diff
@@ -1,23 +1,63 @@
...
+      <p>
+        {weatherData?.name}, {weatherData?.sys?.country}
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa create UI for two cases:
1. weatherData - is present
2. weatherData - is NOT present

File: src/components/Icon.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+const Icon = ({ weatherIconId, alt }) => {
+  return (
+    <img
+      style={{ width: 100, height: 100 }}
+      src={`http://openweathermap.org/img/wn/${weatherIconId}@2x.png`}
+      alt={alt}
+    ></img>
```
Comment (vdanylov) [inline src/components/Icon.js]: @pasternakewa self-closing tag

File: src/components/Icon.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+const Icon = ({ weatherIconId, alt }) => {
+  return (
+    <img
+      style={{ width: 100, height: 100 }}
+      src={`http://openweathermap.org/img/wn/${weatherIconId}@2x.png`}
```
Comment (vdanylov) [inline src/components/Icon.js]: @pasternakewa move to .env

File: src/App.js (hunk 3 above)
Comment (pasternakewa) [reply src/App.js:13]: To fetch only on input submit, not on every input change

File: src/App.js (hunk 7 above)
Comment (pasternakewa) [reply src/App.js]: Could you tell me what exactly should I move to separate function?

File: src/App.js (hunk 8 above)
Comment (pasternakewa) [reply src/App.js]: What exactly please? This button?

File: src/App.js [hunk 12]
```diff
@@ -1,23 +1,91 @@
...
-import './App.css';
+import "./App.css";
+import React, { useState, useEffect } from "react";
+import Icon from "./components/Icon";
+import Input from "./components/Input";
+
+const getQueryUrl = (city) =>
+  `${process.env.REACT_APP_API_URL}?q=${city}&units=metric&appid=${process.env.REACT_APP_API_KEY}`;
+
+const getQueryUrlByCurrentLocation = (currentLocation) => {
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa use destructuring and put values from object directly to the string 
({latitude, longtitude})

File: src/App.js (hunk 7 above)
Comment (vdanylov) [reply src/App.js]: @pasternakewa use just function name
onKeyPress={handleKeypress}

File: src/App.js [hunk 13]
```diff
@@ -1,23 +1,91 @@
...
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
+    <div>
+      <Input
+        value={city}
+        onChange={(e) => setCity(e.target.value)}
+        onKeyPress={(e) => handleKeypress(e)}
+      />
+      <button onClick={() => fetchWeatherData()}>Submit</button>
+      {weatherData && (
```
Comment (vdanylov) [inline src/App.js]: @pasternakewa please fix

**Review by vdanylov — CHANGES_REQUESTED** [review]
Remove empty lines and webviewData check

**Pull request merged**